src/server/req_delete.c

```diff
@@ -394,6 +394,7 @@ req_deletejob(struct batch_request *preq)
 				parent->ji_ajtrk->tkm_tbl[offset].trk_substate =
 					JOB_SUBSTATE_TERMINATED;
 				set_subjob_tblstate(pjob, offset, JOB_STATE_EXPIRED);
+				parent->ji_ajtrk->tkm_dsubjsct++;
```

Comment:
No, In case of IS_ARRAY_Arrayjob, we are deleting the whole array job and it will get purge, so no need to evaluate the state.